woocommerce/woocommerce-ios
# Edit Product: shipping settings UI

Shipping Settings UI for #1422 

## Summary

This PR implements the UI for the Product Shipping Settings. These are the TODO tasks after this PR:
- Fetch the product's shipping class model so that we can display the shipping class name on the Shipping Settings screen
- Navigate to the Shipping Class list selector from the Shipping Settings screen
- Functionality to update the Shipping Settings

## Changes

Note: I will request design review after the shipping class name is displayed

- Created `UnitInputTableViewCell` that has a title, editable text field, and a unit. The UI is configured by `UnitInputViewModel` that contains a protocol `UnitInputFormatter` that formats and validates the unit input
  - Please lemme know if you have a better naming!
- Added some extension helpers to `Product` to create `UnitInputViewModel` for each shipping setting field
- Created `ProductShippingSettingsViewController` that configures 2 sections of 5 fields as in the screenshot below
- In `ProductFormViewController`, added navigation to `ProductShippingSettingsViewController`

## Testing

- Launch the app
- Go to Products tab
- Tap on a Product with non-empty shipping settings
- Tap "Edit"
- Tap "Shiping" row --> the weight, length, width, height, and shipping class slug (if any) should be displayed
- Go back to the Products tab and repeat the above 2 steps with different shipping settings (e.g. empty settings)

## Example screenshots

LTR | RTL
-- | --
![Simulator Screen Shot - iPhone 8 - 2019-12-16 at 16 31 50](https://user-images.githubusercontent.com/1945542/70895362-c4756f80-2029-11ea-9e77-b146ccbb4510.png) | ![Simulator Screen Shot - iPhone 8 - 2019-12-16 at 16 38 40](https://user-images.githubusercontent.com/1945542/70895363-c50e0600-2029-11ea-8b1a-9508bc71aac2.png)


Update release notes:
